sources/identify.php

```diff
@@ -2355,7 +2355,7 @@ function shouldUserAuthWithOauth2(
     // Security issue without this return if an user auth_type == oauth2 and
     // oauth2 disabled : we can login as a valid user by using hashUserId(username)
     // as password in the login the form.
-    if ((int) $SETTINGS['oauth2_enabled'] !== 1) {
+    if ((int) $SETTINGS['oauth2_enabled'] !== 1 && (string) $userInfo['auth_type'] === 'oauth2') {
```

Comment:
Could a local user authenticate via OAuth2 for the first time and "break" their account?